pontoon/base/forms.py

```diff
@@ -158,9 +130,45 @@ def save(self, *args, **kwargs):
         translators = self.cleaned_data.get("translators", User.objects.none())
         managers = self.cleaned_data.get("managers", User.objects.none())
 
+        before_count = self.user.badges_promotion_count
+
         self.assign_users_to_groups("translators", translators)
         self.assign_users_to_groups("managers", managers)
 
+        after_count = self.user.badges_promotion_count
+
+        # Award Community Builder badge
+        if (
+            after_count > before_count
+            and after_count in settings.BADGES_PROMOTION_THRESHOLDS
+        ):
+            self.community_builder_level = (
+                settings.BADGES_PROMOTION_THRESHOLDS.index(after_count) + 1
+            )
+            desc = """
+            You have gained a new badge level!
+            <br>
+            Community Builder Badge: Level {level}
+            <br>
+            You can view this badge on your <a href={profile_href}> profile page </a>.
+            """.format(
+                level=self.community_builder_level,
+                profile_href=reverse(
+                    "pontoon.contributors.contributor.username",
+                    kwargs={
+                        "username": self.user.username,
+                    },
+                ),
+            )
+            notify.send(
+                sender=self.user,
+                recipient=self.user,
+                verb="",  # Triggers render of description only
```

Comment:
Hmmm, this will clash with https://github.com/mozilla/pontoon/pull/3456/files#diff-3e822354a9d9b515f80ad2940a7fb3685084a1ec090224d30fc5ede87f4b0174R458...

I know, the code I'm referencing is brittle, but we should figure something out as a short-term solution. Probably use a unique verb and then handle it when rendering notifications.